[8] GET /todos/:id

pdany1116 opened the pull request:
<!--
How to structure the PR title:
Brief description of your change
-->
#### Description
<!-- Write a brief paragraph that provides the overview and context of the changes being made. -->
Retrieve a single todo via GET /todos/:id.

#### [8](https://trello.com/c/fJJi404t/8-get-todos-id)

#### Related PRs
<!-- Link to any PRs that are related to this one. -->
<!-- A related PR could be changes in a gem that is being update or an API change in another project this PR depends on. -->
<!-- [repo#pull](github link) -->

## Steps to Test or Reproduce
<!-- Write a brief paragraph to explain how to reproduce -->
1. Follow test steps in README.
2. All tests should pass.

## Risks
<!-- Potential risks if any (eg. This PR may affect db structure) -->

pdany1116 committed:
[8] Implement GET /todos/:id

cazacutudor commented on features/todo.feature:
```diff
@@ -6,3 +6,8 @@ Feature: Todo API
     When I retrieve the list of todos
     Then I will get a response with a list containing a todo
+
+  Scenario: Get a todo
```
We should run the integrations specs via CI.

pdany1116 replied:
4d40c5d3473a67df8c118e7ec46d95650ba2bd38

pdany1116 commented on app/controllers/todos_controller.rb:
```diff
@@ -6,4 +6,12 @@ def index
+  def show
+    todo = Todo.find(params[:id])
+
+    render json: todo, status: :ok
```
Serialize the todo response. We do not want to expose sensitive data. To be covered in another ticket.

pdany1116 replied:
https://trello.com/c/7LCo514V/21-serialize-responses

pdany1116 commented on features/support/todo_helpers.rb:
```diff
@@ -5,9 +5,14 @@ module TodoHelpers
   end
+
+  def get_todo(id)
+    get "/todos/#{id}"
```
Use predefined paths by rails.

pdany1116 commented on features/todo.feature:
```diff
@@ -6,3 +6,8 @@ Feature: Todo API
     When I retrieve the list of todos
     Then I will get a response with a list containing a todo
+
+  Scenario: Get a todo
```
Let's add the integration tests in CI.

pdany1116 replied:
Find a solution to run all tests + rubocop.

pdany1116 replied:
4d40c5d3473a67df8c118e7ec46d95650ba2bd38

cazacutudor commented on spec/requests/todos_spec.rb:
```diff
@@ -36,4 +36,33 @@
       end
     end
   end
+
```
We should updates specs accordingly, https://medium.com/just-tech/rspec-controller-or-request-specs-d93ef563ef11

pdany1116 replied:
7545e194cc0b6a41e2d4b9e086c6a336b282e266

pdany1116 committed:
[8] Convert request specs to controller specs

pdany1116 committed:
[8] Use rails predefined route variables

pdany1116 committed:
[8] Use rake to run rspec, cucumber and rubocop

pdany1116 committed:
[8] Disable subject definition cop

pdany1116 committed:
[8] Apply cops for rubocop to pass

pdany1116 committed:
[8] Ignore db schema in rubocop

After db:create db:migrate the schema.rb is overwritten by rails
with a format that rubocop is complaining about.

cazacutudor committed:
[8] Clean up

cazacutudor reviewed (approved):
:rocket:

cazacutudor committed:
[8] Fix rubocop warnings
